explorer-api/src/mix_node/http.rs

```diff
@@ -0,0 +1,124 @@
+                    match get_mix_node_description(
+                        &bond.mix_node.host,
+                        &bond.mix_node.http_api_port,
+                    )
+                    .await
+                    {
+                        Ok(response) => {
+                            // cache the response and return as the HTTP response
+                            state
+                                .inner
+                                .mix_node_cache
+                                .set_description(pubkey.to_string(), response.clone())
+                                .await;
+                            Some(Json(response))
+                        }
+                        Err(e) => {
+                            error!(
+                                "Unable to get description for {} on {}:{} -> {}",
+                                pubkey, bond.mix_node.host, bond.mix_node.http_api_port, e
+                            );
+                            Option::None
+                        }
+                    }
+                }
+                None => Option::None,
+            }
+        }
+    }
+}
+
+#[openapi(tag = "mix_node")]
+#[get("/<pubkey>/stats")]
+pub(crate) async fn get_stats(
+    pubkey: &str,
+    state: &State<ExplorerApiStateContext>,
+) -> Option<Json<NodeStats>> {
+    match state
+        .inner
+        .mix_node_cache
+        .clone()
+        .get_node_stats(pubkey.to_string())
+        .await
+    {
+        Some(cache_value) => {
+            trace!("Returning cached value for {}", pubkey);
+            Some(Json(cache_value))
+        }
+        None => {
+            trace!("No valid cache value for {}", pubkey);
+            match state.inner.get_mix_node(pubkey).await {
+                Some(bond) => {
+                    match get_mix_node_stats(&bond.mix_node.host, &bond.mix_node.http_api_port)
+                        .await
+                    {
+                        Ok(response) => {
+                            // cache the response and return as the HTTP response
+                            state
+                                .inner
+                                .mix_node_cache
+                                .set_node_stats(pubkey.to_string(), response.clone())
+                                .await;
+                            Some(Json(response))
+                        }
+                        Err(e) => {
+                            error!(
+                                "Unable to get description for {} on {}:{} -> {}",
+                                pubkey, bond.mix_node.host, bond.mix_node.http_api_port, e
+                            );
+                            Option::None
+                        }
+                    }
+                }
+                None => Option::None,
+            }
+        }
+    }
+}
+
+async fn get_mix_node_description(host: &str, port: &u16) -> Result<NodeDescription, ReqwestError> {
```

Comment:
I'm surprised clippy didn't yell at you for using `&u16`. You can just use `u16` directly. It's a 2byte primitive value that has `Copy` defined on it and that's cheaper than taking reference.